kore/src/Kore/Exec.hs

```diff
@@ -403,8 +405,8 @@ data StopLabels = StopLabels
 data RpcExecState v = RpcExecState
     { -- | program state
       rpcProgState :: ProgramState (Pattern v)
-    , -- | rule label/id we have stopped on
-      rpcRule :: Maybe Text
+    , -- | rule label/ids we have applied so far
+      rpcRules :: Seq ((UniqueId, Label), Seq UniqueId)
```

Comment:
Why not create a separate datatype for the `((),)`? I think nesting pairs quickly becomes confusing and makes code harder to read. There's also the issue that pairs are lazy by default, would this be helpful in any way? Recall that this backend uses language-strict.

Reply:
sure can change this to a new data type